apps/backend/src/groups/groups.service.ts

```diff
@@ -142,4 +171,46 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  // This method ensures that the passed in user is in all of the
+  // passed in groups, as long as the group already exists.
+  // It will additionally remove the user from any groups not in the list.
+  // Called from oidc.strategy.ts, if OIDC_EXTERNAL_GROUPS is enabled
+  async syncUserGroups(user: User, groups: string[]) {
+    const currentGroups = await user.$get('groups', {include: [User]});
+    const groupsToLeave = currentGroups.filter(
+      (group) => !groups.includes(group.name)
+    );
+    const existingGroups: Group[] = [];
+
+    // Remove user from any groups that they should not be in
+    for (const groupToLeave of groupsToLeave) {
+      try {
+        await this.removeUserFromGroup(groupToLeave, user);
+      } catch (err) {
+        this.logger.warn(`Failed to remove user from group: ${err}`);
+      }
+    }
+
+    // Find existing groups to add user to
+    for (const group of groups) {
+      try {
+        const existingGroup = await this.findByName(group);
+        existingGroups.push(existingGroup);
+      } catch (err) {
+        if (err instanceof NotFoundException) {
+          this.logger.info('External group does not exist locally, skipping..');
+        } else {
+          this.logger.warn(err);
+        }
+      }
+    }
+
+    // Add user to existing groups that they aren't in yet
+    for (const existingGroup of existingGroups) {
+      if (!currentGroups.some((group) => group.name === existingGroup.name)) {
+        this.addUserToGroup(existingGroup, user, 'member');
+      }
+    }
```

Comment:
the addusertogroup promises that are being created in this for loop are not being awaited here nor explicitly returned for the caller to await

could be rewritten in a more functional / less-blocking manner as follows:

```suggestion
    // Add user to existing groups that they aren't in yet
    await Promise.all(existingGroups.filter(existingGroup => !currentGroups.some((group) => group.name === existingGroup.name)).map(this.addUserToGroup(existingGroup, user, 'member')));
```

but you could also just put an `await` in front of the 'addusertogroup' function call

Reply:
based off of the ci this fails cause a) the argument passed into the map isn't a function (we just need to turn it into one) and b) the linter